pages/algorithms/document-context.md

```diff
@@ -0,0 +1,6 @@
+---
+title: Document Context
+key: document-context
+---
+
+Each element has a document root which defines its context. The document context varies if the node is within a `shadowDOM` or perhaps within a nested `iframe`. Other elements usually share the same document context. The document root can usually be identified using the [getRootNode](https://dom.spec.whatwg.org/#ref-for-dom-node-getrootnode) method.
```

Comment:
Remove the "perhaps". Does this apply also to `frames` in a `frameset`? (or do we ignore it because it is not supported in HTML5?)
The text is still a bit ambiguous. What about
Each element has a document context defined by the document root node. A new document root node is created for a `shadowDOM` or a nested `iframe`. The document root can usually be identified using the [getRootNode](https://dom.spec.whatwg.org/#ref-for-dom-node-getrootnode) method.